geomstats/geometry/discrete_surfaces.py

```diff
@@ -0,0 +1,687 @@
+        ----------
+        point: array-like, shape[..., *point_shape]
+            Point.
+
+        Returns
+        -------
+        point: array-like, shape[..., *point_shape]
+            Point.
+        """
+        raise NotImplementedError("projection is not implemented for discrete surfaces")
+
+    def random_point(self, n_samples=1):
+        """Sample discrete surfaces.
+
+        This sample random discrete surfaces with the correct number of vertices.
+
+        Parameters
+        ----------
+        n_samples : int
+            Number of surfaces to sample.
+            Optional, Default=1
+
+        Returns
+        -------
+        vertices :  array-like, shape=[n_samples, n_vertices, 3]
+            Vertices for a batch of points in the space of discrete surfaces.
+        """
+        sphere = Hypersphere(dim=2, default_coords_type="extrinsic")
+        vertices = sphere.random_uniform(n_samples * self.n_vertices)
+        vertices = gs.reshape(vertices, (n_samples, self.n_vertices, 3))
+        return vertices
+
+    def vertex_areas(self, point):
+        """Compute vertex areas for a triangulated surface.
+
+        Parameters
+        ----------
+        point : array-like, shape=[n_verticesx3]
+             Surface, i.e. the vertices of its triangulation.
+
+        Returns
+        -------
+        vertareas :  array-like, shape=[n_verticesx1]
+            vertex areas
+        """
+        n_vertices = point.shape[0]
+        face_coordinates = point[self.faces]
+        v0, v1, v2 = (
+            face_coordinates[:, 0],
+            face_coordinates[:, 1],
+            face_coordinates[:, 2],
+        )
+        A = gs.linalg.norm((v1 - v2), axis=1)
+        B = gs.linalg.norm((v0 - v2), axis=1)
+        C = gs.linalg.norm((v0 - v1), axis=1)
+        s = 0.5 * (A + B + C)
+        area = gs.sqrt((s * (s - A) * (s - B) * (s - C)).clamp(min=1e-6))
+        idx = gs.flatten(self.faces)
+        incident_areas = gs.zeros(n_vertices)
+        val = gs.flatten(gs.stack([area] * 3, axis=1))
+        incident_areas.scatter_add_(0, idx, val)
+        vertAreas = 2 * incident_areas / 3.0
+        return vertAreas
+
+    def get_laplacian(self, point):
+        """Compute the mesh Laplacian operator of a surface.
+
+        The laplacian is evaluated at one of its tangent vectors h.
+
+        Parameters
+        ----------
+        point  :  array-like, shape=[n_verticesx3]
+             Surface, i.e. the vertices of its triangulation.
+
+        Returns
+        -------
+        L : callable
+            Function that will evaluate the mesh Laplacian operator
+            at a tangent vector to the surface
+        """
+        n_vertices, n_faces = point.shape[0], self.faces.shape[0]
+        face_coordinates = point[self.faces]
+        v0, v1, v2 = (
+            face_coordinates[:, 0],
+            face_coordinates[:, 1],
+            face_coordinates[:, 2],
+        )
+        A = gs.linalg.norm((v1 - v2), axis=1)
```

Comment:
Ditto

Reply:
not letting me resolve, but this is done